packages/roosterjs-editor-plugins/lib/plugins/ImageSelection/ImageSelection.ts

```diff
@@ -37,35 +49,52 @@ export default class ImageSelection implements EditorPlugin {
     dispose() {
         this.editor.select(null);
         this.editor = null;
-        this.imageId = null;
     }
 
     onPluginEvent(event: PluginEvent) {
         if (this.editor) {
             switch (event.eventType) {
                 case PluginEventType.EnteredShadowEdit:
+                case PluginEventType.LeavingShadowEdit:
                     const selection = this.editor.getSelectionRangeEx();
                     if (selection.type == SelectionRangeTypes.ImageSelection) {
                         this.editor.select(selection.image);
                     }
                     break;
-                case PluginEventType.LeavingShadowEdit:
-                    if (this.imageId) {
-                        const images = this.editor.queryElements(
-                            '#' + this.imageId
-                        ) as HTMLImageElement[];
-                        if (images.length == 1) {
-                            const image = images[0];
-                            this.editor.select(image);
-                        }
-                        this.imageId = null;
-                    }
-                    break;
+
                 case PluginEventType.MouseDown:
                     const target = event.rawEvent.target;
                     if (safeInstanceOf(target, 'HTMLImageElement')) {
                         this.editor.select(target);
-                        this.imageId = target.id;
+                    }
+                    break;
+                case PluginEventType.KeyDown:
+                    const key = event.rawEvent.key;
+                    const keyDownSelection = this.editor.getSelectionRangeEx();
+                    if (keyDownSelection.type === SelectionRangeTypes.ImageSelection) {
+                        if (
+                            key === Escape ||
+                            key === PageDown ||
+                            key === PageUp ||
+                            key === Home ||
+                            key === End
+                        ) {
+                            this.editor.select(keyDownSelection.image, PositionType.Before);
+                            this.editor.getSelectionRange().collapse();
+                        } else if (
+                            key === SHIFT ||
+                            key === CTRL ||
+                            key === Alt ||
+                            key === Enter ||
+                            key == Space ||
+                            event.rawEvent.metaKey
+                        ) {
+                            this.editor.select(keyDownSelection.ranges[0]);
+                        } else {
+                            keyDownSelection.ranges[0].deleteContents();
+                            const cursorPosition = this.editor.getSelectionRange();
```

Comment:
Actually, can we just allow the behavior we have in ImageEdit plugin here https://github.com/microsoft/roosterjs/blob/fc0408cf3ce3165fc9625b4a1ae1beb3026f3a78/packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts#L205 to handle keyboard event?

Reply:
In most cases (actually almost all), we just need to select the image with normal selection, then let browser handle the rest.